src/main/scala/rover/rdo/ObjectState.scala

```diff
@@ -1,29 +1,26 @@
 package rover.rdo
-
-import votingapp.{PollChoice, Votes}
-
+	
 trait ObjectState {
 
 }
 
-class AtomicObjectState extends ObjectState {
+abstract class AtomicObjectState[A <: AtomicObjectState[A]] extends ObjectState {
+	// binding this to a self-type A
+	this: A =>
+	
 	// not sure due to types... do we want the _specific_ subtype here? Not the abstract state
-	private var ops: List[Op[AtomicObjectState]] = List()
+	private var ops: List[Op] = List()
```

Comment:
They would we `Op`s over `A` right?

Reply:
Yes, Op is defined as a function type with the following signature: `A =>A`